src/libcore/fmt/builders.rs

```diff
@@ -165,6 +167,72 @@ impl<'a, 'b: 'a> DebugStruct<'a, 'b> {
         self
     }
 
+    /// Add a field to the struct's debug format, but show the value of that field to be
+    /// unrepresentable.
+    ///
+    /// This method is useful if you want to show that a struct has a field, but you cannot get a
+    /// debug representation of that field.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(debug_non_exhaustive)]
+    /// use std::fmt;
+    ///
+    /// struct Foo;
```

Comment:
I'd add a field `foo` to this structure, just for clarity's sake.

Probably ``Box<dyn (pick your favorite object safe trait that isn't `Debug`)>``.